server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -730,10 +730,24 @@ private Symbol resolveUnindexedSubscriptExpression(
                 throw e;
             }
             try {
-                Symbol base = fieldProvider.resolveField(qualifiedName,
-                    List.of(),
-                    operation,
-                    context.errorOnUnknownObjectKey());
+                Symbol base = null;
+                // From the perspective of the outer query for `select o['a']['b'] from (select o['a'] from t) t2`,
+                // the base column of `o['a']['b']` could be `o` or `o['a']`.
+                for (int i = parts.size() - 1; i >= 0; i--) {
+                    try {
+                        base = fieldProvider.resolveField(qualifiedName,
+                            parts.subList(0, i),
+                            operation,
+                            context.errorOnUnknownObjectKey());
+                        if (base != null) {
+                            break;
+                        }
+                    } catch (ColumnUnknownException ignored) {
+                    }
```

Comment:
Shouldn't this result in an exception if `context.errorOnUnknownObjectKey()`? I mean, do we need to catch this here?

Reply:
Thanks a lot for reviewing. As you said I think we don't need to catch when `context.errorOnUnknownObjectKey()`.

Reply:
I tried but could not find a case where we need to catch the exception when `error_on_unknown_object_key = true`. Please check [9627f9b](https://github.com/crate/crate/pull/17149/commits/9627f9bbeeef9882db96572f0fc7a59197ac42db). I think we could also do something like:
```
if (context.errorOnUnknownObjectKey()) {
  // original logic
} else {
for (int i = parts.size() - 1; i >= 0; i--) {
  try {
      base = fieldProvider.resolveField(qualifiedName,
          parts.subList(0, i),
          operation,
          context.errorOnUnknownObjectKey());
      if (base != null) {
          break;
      }
  } catch (ColumnUnknownException ignored) {
  }
}
```

Reply:
Hi @seut could you review this again? Now the new logic only applies when `context.errorOnUnknownObjectKey()` is `false`.

Reply:
@jeeminso Please hold on with this change at all as I think we should
- Merge https://github.com/crate/crate/pull/17139 first
- Try to tackle the root causes of the issues which in this case is the issue that we treat the alias mapping maybe wrongly. E.g. `select obj_dynamic['u']['u2'] from (select obj_dynamic['u'] from t) t2;` is strictly speaking selecting `select obj_dynamic['u'] ['u']['u2'] from t` as the return col from the sub-select is already `obj_dynamic['u']`.